networkit/cpp/independentset/Luby.cpp

```diff
@@ -42,28 +44,33 @@ std::vector<bool> Luby::run(const Graph &G) {
     while (!empty()) {
         // choose set S - weighted choice of active nodes with probability $1 / 2 \omega(v)$
         std::vector<bool> S(G.numberOfNodes(), false);
+
         G.parallelForNodes([&](node u) {
             if (V[u]) {
                 if (Aux::Random::probability() < nodeProbability(u)) {
                     S[u] = true; // add node to S
                 }
             }
         });
+
         // remove non-independent nodes from S to get S'
-        G.parallelForEdges([&](node u, node v) {
-            if (u != v) {           // exclude self-loops
-                if (S[u] && S[v]) { // u and v are not independent (note: S is subset of V')
-                    // remove node with smaller degree
-                    edgeweight wu = weightedDegree(u);
-                    edgeweight wv = weightedDegree(v);
-                    if (wu > wv) {
-                        S[v] = false;
-                    } else if (wv > wu) {
-                        S[u] = false;
-                    } else {          // tie
-                        S[v] = false; // arbitrary decision
+        G.parallelForNodes([&](node u) {
+            if (S[u]) {
```

Comment:
You can avoid an indentation level:
```suggestion
            if (!S[u]) return;
```